test/unit/client-href-builder.js

```diff
@@ -87,6 +87,26 @@ describe('client : href-builder :', () => {
       });
     });
 
+    describe('when some of the paramaters are undefined or null', () => {
```

Comment:
Actually, `null` is different from `undefined`. `null` is valid in JSON and we should keep it blank if it's there. I believe the only thing we should tackle here is `undefined`.

Reply:
NaN